--- a/scala/private/common.bzl
+++ b/scala/private/common.bzl
@@ -1,3 +1,4 @@
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("//scala:jars_to_labels.bzl", "JarsToLabelsInfo")
 load("//scala:plusone.bzl", "PlusOneDeps")
 load("//scala:providers.bzl", "ScalaInfo")
--- a/scala/private/rules/scala_library.bzl
+++ b/scala/private/rules/scala_library.bzl
@@ -1,3 +1,4 @@
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//lib:dicts.bzl", _dicts = "dicts")
 load(
     "//scala/private:common.bzl",
--- a/scala/private/rules/scala_doc.bzl
+++ b/scala/private/rules/scala_doc.bzl
@@ -1,4 +1,4 @@
 """Scaladoc support"""
 
-load("//scala/private:common.bzl", "collect_plugin_paths")
+load("//scala/private:common.bzl", "JavaInfo", "collect_plugin_paths")
 load("//scala:providers.bzl", "ScalaInfo")
--- a/scala/private/rules/scala_binary.bzl
+++ b/scala/private/rules/scala_binary.bzl
@@ -1,3 +1,4 @@
 """Builds Scala binaries"""
 
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//lib:dicts.bzl", _dicts = "dicts")
--- a/scala/private/rules/scala_test.bzl
+++ b/scala/private/rules/scala_test.bzl
@@ -1,3 +1,4 @@
 """Rules for writing tests with ScalaTest"""
 
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//lib:dicts.bzl", _dicts = "dicts")
--- a/scala/private/rules/scala_junit_test.bzl
+++ b/scala/private/rules/scala_junit_test.bzl
@@ -1,3 +1,4 @@
 """Rules for writing tests with JUnit"""
 
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//lib:dicts.bzl", _dicts = "dicts")
--- a/scala/private/rules/scala_repl.bzl
+++ b/scala/private/rules/scala_repl.bzl
@@ -1,3 +1,4 @@
 """Rule for launching a Scala REPL with dependencies"""
 
+load("@rules_java//java/common:java_info.bzl", "JavaInfo")
 load("@bazel_skylib//lib:dicts.bzl", _dicts = "dicts")
